### PR TITLE
Use Assert instead of errorBuilder in MessageContractTests.cs

### LINKED ISSUE #672: Use Assert instead of errorBuilder in MessageContractTests.cs
We should use normal Assert pattern as we are moving away from errorBuilder.

### PR DESCRIPTION
**Pull request by shmao**
Fixes #672

### CODE REVIEW

**Comment by shmao** [comment]
@dotnet-bot test outerloop please

File: src/System.Private.ServiceModel/tests/Scenarios/Contract/Message/MessageContractTests.cs [hunk 1]
```diff
@@ -33,20 +31,14 @@ public static void MessageContract_IsWrapped_True()
...
         XmlDictionaryReader reader = MessageContractHelpers.GetResponseBodyReader(inspector);
 
-        if (reader.LocalName.Equals(MessageContractConstants.wrapperName))
-        {
-            errorBuilder.AppendLine(String.Format("When IsWrapped set to false, the message body should not be wrapped with an extra element."));
-        }
-
-        Assert.True(errorBuilder.Length == 0, errorBuilder.ToString());
+        Assert.False(reader.LocalName.Equals(MessageContractConstants.wrapperName),
+        "When IsWrapped set to false, the message body should not be wrapped with an extra element.");
```
Comment (iamjasonp) [inline src/System.Private.ServiceModel/tests/Scenarios/Contract/Message/MessageContractTests.cs]: NIT: indent second line

File: src/System.Private.ServiceModel/tests/Scenarios/Contract/Message/MessageContractTests.cs [hunk 2]
```diff
@@ -117,7 +107,8 @@ public static void MessageBody_Elements_CustomerElement_Value_Matches()
                     }
                     else
                     {
-                        errorBuilder.AppendLine(String.Format("Comparison Failed. Expected: {0}, Actual: {1}", MessageContractConstants.customerElementValue, reader.Value));
+                        Assert.True(false, 
+                            string.Format("Comparison Failed. Expected: {0}, Actual: {1}", MessageContractConstants.customerElementValue, reader.Value));
```
Comment (iamjasonp) [inline src/System.Private.ServiceModel/tests/Scenarios/Contract/Message/MessageContractTests.cs]: This should be replaceable with Assert.Equal, as that will also output the expected and actual value - it does away with the string.Format entirely

File: src/System.Private.ServiceModel/tests/Scenarios/Contract/Message/MessageContractTests.cs [hunk 3]
```diff
@@ -125,17 +116,13 @@ public static void MessageBody_Elements_CustomerElement_Value_Matches()
...
-            {
-                errorBuilder.AppendLine(String.Format("Expected element not found. Looking For: {0} && {1}", MessageContractConstants.customerElementName, MessageContractConstants.customerElementNamespace));
-            }
+            Assert.True(elementFound,
+                string.Format("Expected element not found. Looking For: {0} && {1}", MessageContractConstants.customerElementName, MessageContractConstants.customerElementNamespace));
         }
         catch (Exception ex)
         {
-            errorBuilder.AppendLine(String.Format("Unexpected exception was caught: {0}", ex.ToString()));
+            Assert.True(false, string.Format("Unexpected exception was caught: {0}", ex.ToString()));
```
Comment (iamjasonp) [inline src/System.Private.ServiceModel/tests/Scenarios/Contract/Message/MessageContractTests.cs]: If you remove the catch block, no need to Assert.True(false) here.

**Comment by iamjasonp** [comment]
Just some nits here and there, else the tests LGTM given there are no substantive changes to the tests themselves, just the Assert vs. StringBuilder logic.

File: src/System.Private.ServiceModel/tests/Scenarios/Contract/Message/MessageContractTests.cs (hunk 2 above)
Comment (shmao) [inline src/System.Private.ServiceModel/tests/Scenarios/Contract/Message/MessageContractTests.cs]: You are right. Made a change on this.

**Comment by shmao** [comment]
@dotnet-bot test outerloop please

**Comment by shmao** [comment]
@dotnet-bot test outerloop please

**Comment by shmao** [comment]
The outerloop test failed due to the failure of test HttpsTests.ServerCertificateValidation_EchoString, which seems to be a known issue. @roncain can I merge this PR in this case?

**Comment by iamjasonp** [comment]
ACK to merge given we know about HttpsTests.ServerCertificateValidation_EchoString

**Pull request merged**